src/azul/service/repository_controller.py

```diff
@@ -61,7 +61,12 @@ def download_file(self,
             file_version = file['version']
             drs_path = file['drs_path']
             if file_name is None:
-                file_name = file['name']
+                # To prevent file overwrites due to files with the same name
+                # we insert the crc32c value of the file into the filename.
+                parts = file['name'].split('.')
+                append_to = -3 if parts[-1] == 'gz' and len(parts) > 2 else -2
```

Comment:
Not what I had in mind when I said "maintain a list of composite extensions" yesterday. Feel free to ask for clarification on parking lot.